src/ImageSharp/Formats/Tga/TgaEncoderCore.cs

```diff
@@ -225,35 +228,29 @@ private void WriteRunLengthEncodedImage<TPixel>(Stream stream, ImageFrame<TPixel
         /// <typeparam name="TPixel">The pixel type.</typeparam>
         /// <param name="pixels">The pixels of the image.</param>
         /// <param name="xStart">X coordinate to start searching for the same pixels.</param>
-        /// <param name="yStart">Y coordinate to start searching for the same pixels.</param>
+        /// <param name="yPos">Y coordinate to searching for the same pixels in only one scan line.</param>
         /// <returns>The number of equal pixels.</returns>
-        private byte FindEqualPixels<TPixel>(Buffer2D<TPixel> pixels, int xStart, int yStart)
+        private byte FindEqualPixels<TPixel>(Buffer2D<TPixel> pixels, int xStart, int yPos)
             where TPixel : unmanaged, IPixel<TPixel>
         {
             byte equalPixelCount = 0;
-            bool firstRow = true;
-            TPixel startPixel = pixels[xStart, yStart];
-            for (int y = yStart; y < pixels.Height; y++)
+            TPixel startPixel = pixels[xStart, yPos];
+            for (int x = xStart + 1; x < pixels.Width; x++)
             {
-                for (int x = firstRow ? xStart + 1 : 0; x < pixels.Width; x++)
+                TPixel nextPixel = pixels[x, yPos];
+                if (startPixel.Equals(nextPixel))
+                {
+                    equalPixelCount++;
+                }
+                else
                 {
-                    TPixel nextPixel = pixels[x, y];
-                    if (startPixel.Equals(nextPixel))
-                    {
-                        equalPixelCount++;
-                    }
-                    else
-                    {
-                        return equalPixelCount;
-                    }
-
-                    if (equalPixelCount >= 127)
-                    {
-                        return equalPixelCount;
-                    }
+                    return equalPixelCount;
                 }
 
-                firstRow = false;
+                if (equalPixelCount >= 127)
```

Comment:
Reading the spec again, I believe this should be >=128 instead.

From the spec page 25: ```1. Pixel Count: The pixel count can range from 0 to 127. Since a run-length packet never encodes
zero pixels, and in order to make use of all of the values available with seven bits, a zero (0) in Pixel Count
is defined to mean that 1 pixel is encoded by the packet. A 1 in Pixel Count means that 2 pixels are
actually contained in the packet. In other words, the value in Pixel Count is always 1 less than the actual
number of pixels encoded in the packet. Thus, you may encode between 1 and 128 pixels with a single
packet. If you have a run which is longer than 128 bytes, you must encode it using multiple packets.```

This highlights two more problems: 
1. **run-length packet never encodes zero pixels**, which we dont respect. We should add a case for this.
2. **1 in Pixel Count means that 2 pixels are actually contained in the packet.** I believe we are not doing this correct either.

Reply:
Mhm I think, I miss-judged this yesterday. After reading it once again, I think finding equal pixel count is fine with >= 127.
The loop starts with `x = xStart + 1`, which means if we find 127 equal pixels, with the first pixel we encode 128 in one run at maximum.

Also we do follow **run-length packet never encodes zero pixels**: If `equalPixelCount` is zero, the run contains 1 pixel.

I was irritated about the difference about our encoded data and the one from imagemagick and thought this was the cause. Sorry for the confusion.